src/passkey_child/passkey_child_assert.c

```diff
@@ -75,28 +75,59 @@ get_assert_user_id(TALLOC_CTX *mem_ctx, fido_assert_t *assert,
 }
 
 errno_t
-set_assert_client_data_hash(fido_assert_t *_assert)
+set_assert_client_data_hash(const struct passkey_data *data,
+                            fido_assert_t *_assert)
 {
+    TALLOC_CTX *tmp_ctx = NULL;
     unsigned char cdh[32];
+    unsigned char *crypto_challenge = NULL;
+    size_t crypto_challenge_len;
     errno_t ret;
 
-    ret = sss_generate_csprng_buffer(cdh, sizeof(cdh));
-    if (ret != EOK) {
-        DEBUG(SSSDBG_OP_FAILURE,
-              "sss_generate_csprng_buffer failed [%d]: %s.\n",
-              ret, fido_strerr(ret));
-        goto done;
+    tmp_ctx = talloc_new(NULL);
+    if (tmp_ctx == NULL) {
+        DEBUG(SSSDBG_OP_FAILURE, "talloc_new() failed.\n");
+        return ENOMEM;
     }
 
-    ret = fido_assert_set_clientdata_hash(_assert, cdh, sizeof(cdh));
-    if (ret != FIDO_OK) {
-        DEBUG(SSSDBG_OP_FAILURE,
-              "fido_assert_set_clientdata_hash failed [%d]: %s.\n",
-              ret, fido_strerr(ret));
-        goto done;
+    if (data->action == ACTION_AUTHENTICATE) {
+        ret = sss_generate_csprng_buffer(cdh, sizeof(cdh));
```

Comment:
Why do you generate random data which is then overwritten by fido_assert_set_clientdata_hash?

Reply:
I think you misunderstood the purpose of `fido_assert_set_clientdata_hash()`. Let's keep the discussion in the previous conversation.

Reply:
Right, I understood it from the function documentation as cdh is coppied to assert.